src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -197,6 +208,26 @@ library BeaconChainProofs {
         require(Merkle.verifyInclusionSha256(proof, beaconStateRoot, balanceRoot, balanceIndex), "BeaconChainProofs.verifyValidatorBalance: Invalid merkle proof");
     }
 
+    function verifySlotRoot(
+        bytes32 beaconStateRoot,
+        bytes32 slotRoot,
+        bytes calldata proof
+    ) internal view {
+        require(proof.length == 32 * (BEACON_STATE_FIELD_TREE_HEIGHT), "BeaconChainProofs.verifySlotRoot: Proof has incorrect length");
+        //Next we verify the slot against the blockHeaderRoot
+        require(Merkle.verifyInclusionSha256(proof, beaconStateRoot, slotRoot, BEACON_STATE_SLOT_INDEX), "BeaconChainProofs.verifyWithdrawalProofs: Invalid slot merkle proof");
+    }
+
+    function verifyStateRootAgainstLatestBlockHeaderRoot(
```

Comment:
does this actually have to be the *latest*? doesn't it work for any block header root?

Reply:
discussed this offline, supposedly the field in the container is literally named something like `latestBlockHeader`.
Sounds like some cursed naming to me; we should point specifically to where in the Beacon Chain spec this is defined so it's clear where the name is coming from, because I find it confusing (as does G)